Source/Windows/ARM64EC/Module.cpp

```diff
@@ -196,6 +194,31 @@ void PatchCallChecker() {
   WriteModuleRVA(Module, CHPEMetadata->__os_arm64x_dispatch_icall_cfg, &CheckCall);
 }
 
+// Fills in the syscall numbers necessary to call *Native variants of syscalls from FEX under wine.
+void ParseWineSyscallNumbers(HMODULE NtDll) {
+  ULONG Size;
+  const auto* Exports = reinterpret_cast<IMAGE_EXPORT_DIRECTORY*>(RtlImageDirectoryEntryToData(NtDll, true, IMAGE_DIRECTORY_ENTRY_EXPORT, &Size));
+  const auto* NameTable = reinterpret_cast<uint32_t*>(NtDllBase + Exports->AddressOfNames);
+
+  // Wine orders syscalls alphabetically, take advantage of that to find the syscall indices for those which we need to
+  // manually issue. Note that all functions starting with Nt besides NtGetTickCount are syscalls.
+  uint32_t CurSyscallId = 0;
+  for (uint32_t Idx = 0; Idx < Exports->NumberOfNames; Idx++) {
+    const char* Name = reinterpret_cast<const char*>(NtDllBase + NameTable[Idx]);
+    if (Name[0] == 'N' && Name[1] == 't' && strcmp(Name, "NtGetTickCount")) {
```

Comment:
Does this strcmp against `NtGetTickCount` actually do anything since you're going to immediately follow it up with an exact name check against the three syscalls we care about?
I can understand checking that the first two characters are `Nt` at least.

Reply:
It's done so that CurSyscallId isn't incremented for it

Reply:
ah! I see. I misread the code. Good to know